database/sql/src/main/scala/cromwell/database/slick/CallCachingSlickDatabase.scala

```diff
@@ -92,12 +92,31 @@ trait CallCachingSlickDatabase extends CallCachingSqlDatabase {
     runTransaction(action)
   }
 
-  override def cacheEntryExistsForCall(workflowExecutionUuid: String, callFqn: String, index: Int)
-                                      (implicit ec: ExecutionContext): Future[Boolean] = {
+  private def callCacheJoinFromEntryQuery(callCachingEntry: CallCachingEntry)
+                            (implicit ec: ExecutionContext) = {
+    val callCachingEntryId = callCachingEntry.callCachingEntryId.get
+    for {
+      callCachingSimpletonEntries <- dataAccess.
+        callCachingSimpletonEntriesForCallCachingEntryId(callCachingEntryId).result
+      callCachingDetritusEntries <- dataAccess.
+        callCachingDetritusEntriesForCallCachingEntryId(callCachingEntryId).result
+      callCachingHashEntries <- dataAccess.
+        callCachingHashEntriesForCallCachingEntryId(callCachingEntryId).result
+      callCachingAggregationEntries <- dataAccess.
+        callCachingAggregationForCacheEntryId(callCachingEntryId).result.headOption
+    } yield CallCachingJoin(callCachingEntry, callCachingHashEntries, callCachingAggregationEntries, callCachingSimpletonEntries, callCachingDetritusEntries)
+  }
+
+  override def callCacheJoinForCall(workflowExecutionUuid: String, callFqn: String, index: Int)
+                                   (implicit ec: ExecutionContext): Future[Option[CallCachingJoin]] = {
     val action = for {
       callCachingEntryOption <- dataAccess.
         callCachingEntriesForWorkflowFqnIndex((workflowExecutionUuid, callFqn, index)).result.headOption
-    } yield callCachingEntryOption.nonEmpty
+      callCacheJoin <- callCachingEntryOption match {
+        case Some(entry) => callCacheJoinFromEntryQuery(entry).map(Option.apply)
```

Comment:
ToL I've been spreading the good word about `cats.syntax.option._`'s `fold` method which basically does this match but with less boilerplate

Reply:
Leaving aside the "what is this, amateur hour?" meme, I've been personally swayed back to the `match` structure via vigorous support from Odersky (who also notes that it's the fastest scheme out there, but premature optimization and all of that).

I don't really care either way but throwing it out there

Reply:
Style over speed IMO.  

actually now that I think about it, this is a `map(callCacheJoinFromEntryQuery).getOrElse(DBIO.successful(None))` in disguise

Reply:
matcheroni

Reply:
I want to like fold for options, but never have. I did live squarely in team flatmap for a while though.